roles/StackStorm.st2smoketests/tasks/main.yml

```diff
@@ -41,12 +41,14 @@
   tags:
     - smoke-tests
 
-- name: Install st2 pack from exchange
-  become: yes
+- name: Get st2 pack info
   environment:
     ST2_AUTH_TOKEN: "{{ st2_token.stdout }}"
-  command: "st2 pack install st2"
+  command: st2 pack get st2 -j
   changed_when: no
+  check_mode: no
+  register: _smoke_st2_pack_installed
+  failed_when: _smoke_st2_pack_installed.rc != 0
```

Comment:
The Ansible task should already fail if it exits with non-zero code,
this check looks excessive.

Reply:
Fine :)